lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.h

```diff
@@ -946,6 +958,8 @@ class SwiftASTContext : public TypeSystemSwift {
       library_load_cache;
   /// A cache for GetCompileUnitImports();
   llvm::DenseSet<std::pair<Module *, lldb::user_id_t>> m_cu_imports;
+  /// clang module cache path.
+  std::string m_module_cache_path = "";
```

Comment:
Why do we need to store this separately? Isn't it sufficient to compute it once and then only have it stored in the CompilerInvocation?

Reply:
This is prevents the path gets computed every time a context is requested. Is there a better place for it or you prefer to re-compute every time?

Reply:
Each SwiftASTContext owns exactly one CompilerInvocation, there is nothing gained by storing m_module_cache_path in SwiftASTContext.

Reply:
But you can sink the code that generates the tmpdir into GetClangModulesCacheProperty, and potentially avoid running that if GetClangModulesCachePath didn't change (by storing the last result in a static variable).

Reply:
I think I figured out. Let me know what you think now.